### ISSUE TITLE
pachctl inspect job may decrease the number of processed datums reported as complete

### ISSUE THREAD

**Issue by RaananHadar** [pachyderm/pachyderm#6144]
**What happened?**:
I ran `watch -n 0.1 pachctl inspect job <job_id>` and saw that the number of processed datum can also decrease.

**What you expected to happen?**:
The reported number should be monotonically rising.

**How to reproduce it (as minimally and precisely as possible)?**:
This may be challenging to produce without my pipeline but the pipeline finished processing datums every 2 seconds or so (relatively rapidly)

**Environment?**:
- Kubernetes version (use `kubectl version`): AKS 1.20.5
- Pachyderm CLI and pachd server version (use `pachctl version`): 1.13.2
- Cloud provider (e.g. aws, azure, gke) or local deployment (e.g. minikube vs dockerized k8s): azure

**Comment by dgeorg42** [comment]
There are certain scenarios in which this can happen, @RaananHadar.  You won't see it happen with `list job`, though, only with `inspect job`.  Here's an example to illustrate:

Let's say that you have the following for a particular job run:
- 500 datums to be processed;
- 1 worker to process them (to keep the example simple); and
- chunk_spec configured in such a way that the worker gets 20 datums at a time.

First let's talk about `list job`. The `list job` command reports the number of datums processed based upon what's been committed by the workers.  The worker will commit its work at the conclusion of the chunk it is processing. So if you run `list job` repeatedly over the course of the job running, you'll see the number of datums processed incrementing by 20s.

`inspect job` on the other hand looks at what's been committed, but then also queries the workers to find out how far along each of them is in their current chunks. So let's say that 400 datums have been committed, and your single worker has processed 16 of the 20 datums in the current chunk.  `inspect job` would show 416 datums processed (versus the 400 that `list job` would show).  

Now, that doesn't explain why the number processed would go down, but..... Let's say that after processing the 16th datum in the chunk, the worker crashes and restarts.  After restarting, the worker begins to re-process the chunk that it was on.  But the worker is actually smart enough to know that the first 16 datums were already processed, so it skips them and then moves on to process datums 17, 18, 19, and 20.  But let's say you run `list job` and `inspect job` just after the worker processes the 19th datum.  At that point `list job` will still show 400 (the total fully committed) whereas `inspect job` will now only show 403 datums processed, a decrease from the 416 it showed before. And that's because when the worker is queried it will respond that it has skipped 16 datums (1-16) and processed 3 datums (17, 18, and 19). 

The bottom line is that `list job` will always give you an answer that is 100% correct - the number of datums processed and fully committed - whereas `inspect job` will give you a more granular answer of the number processed, but it may or may not be 100% accurate.

**Comment by RaananHadar** [comment]
Thank  you for the very insightful and detailed answer @dgeorg42 , closing.
